packages/footer/components/Newsletter.tsx

```diff
@@ -0,0 +1,239 @@
+   inputRef,
+   placeholder,
+   buttonText,
+   isSubscribed,
+   subscription,
+   onSubmit,
+}: {
+   inputRef: React.RefObject<HTMLInputElement>;
+   placeholder: string;
+   buttonText: string;
+   isSubscribed: boolean;
+   subscription: Subscription;
+   onSubmit: (event: React.FormEvent<HTMLElement>) => Promise<void>;
+}) => {
+   if (isSubscribed) {
+      return null;
+   }
+   return (
+      <HStack
+         as="form"
+         onSubmit={onSubmit}
+         data-testid="footer-newsletter-form"
+         spacing="3"
+         w={{
+            base: 'full',
+            xl: 'auto',
+         }}
+         justify={{
+            base: 'center',
+            md: 'flex-end',
+         }}
+         align="flex-start"
+      >
+         <NewsletterInput
+            inputRef={inputRef}
+            placeholder={placeholder}
+            subscription={subscription}
+         />
+         <Button
+            type="submit"
+            data-testid="footer-newsletter-subscribe-button"
+            isLoading={subscription.status === SubscriptionStatus.Subscribing}
+            disabled={subscription.status !== SubscriptionStatus.Idle}
+            bg="blue.ifixit"
+            color="white"
+            border="none"
+            fontWeight="bold"
+            _hover={{ bg: 'blue.ifixit', boxShadow: '0 0 5px gray' }}
+         >
+            {buttonText}
+         </Button>
+      </HStack>
+   );
+};
+
+const NewsletterSubscribed = ({ isSubscribed }: { isSubscribed: boolean }) => {
+   if (!isSubscribed) {
+      return null;
+   }
+   return (
+      <Flex
+         height="42px"
+         flexDirection="row"
+         alignItems="center"
+         padding="12px"
+         gap="8px"
+         w="100%"
+         backgroundColor="#EDF6FF"
+         border="1px"
+         borderColor="#77B5FF"
+         borderStyle="solid"
+         borderRadius="4px"
+         mt="0px !important"
+      >
+         <FaIcon icon={faCircleCheck} color="#1975F1" />
+         <Text color="gray.900" m={0} fontSize="16px">
+            Subscribed!
+         </Text>
+      </Flex>
+   );
+};
+
+const NewsletterLink = ({ isSubscribed }: { isSubscribed: boolean }) => {
+   const message = isSubscribed ? 'Let me read it!' : 'Let me read it first!';
+   return (
+      <Link
+         href="/Newsletter"
+         color="gray.300"
+         marginTop={2}
+         textDecoration="underline"
+         _hover={{ color: 'white' }}
+         _visited={{ color: 'gray.300' }}
+      >
+         <FaIcon icon={faEye} display="inline" mr={1} verticalAlign="-3px" />
+         {message}
+      </Link>
+   );
+};
+
+export function NewsletterComponent({
+   newsletterForm,
+}: {
+   newsletterForm: NewsletterFormProps;
+}) {
+   const ref = React.useRef(null);
+   const inputRef = React.useRef<HTMLInputElement>(null);
+   const [subscription, subscribe] = useSubscribeToNewsletter();
+
+   const onSubscribe = React.useCallback(
+      async (event: React.FormEvent<HTMLElement>) => {
```

Comment:
This doesn't use `await` and thus doesn't need to be declared as `async`

Reply:
I actually think it'd be cleaner to move the Ref stuff and this event handler down into the `NewsletterForm` and have it just take in `onSubscribe: (email: string) => void`. Mostly because I think the handler should be in the component that is responsible for the form. Then this component can just do `onSubscribe={subscribe}`.

Reply:
Nice idea! Moved in [0389595](https://github.com/iFixit/react-commerce/pull/1342/commits/0389595395eb69a7a005f9a73c7bd123951b66c8)